plan/typeinferer.go

```diff
@@ -299,20 +299,28 @@ func (v *typeInferrer) handleFuncCallExpr(x *ast.FuncCallExpr) {
 	)
 	switch x.FnName.L {
 	case ast.Abs, ast.Ifnull, ast.Nullif:
-		tp = x.Args[0].GetType()
-		// TODO: We should cover all types.
-		if x.FnName.L == ast.Abs && tp.Tp == mysql.TypeDatetime {
-			tp = types.NewFieldType(mysql.TypeDouble)
+		if len(x.Args) > 0 {
+			tp = x.Args[0].GetType()
+			// TODO: We should cover all types.
+			if x.FnName.L == ast.Abs && tp.Tp == mysql.TypeDatetime {
+				tp = types.NewFieldType(mysql.TypeDouble)
+			}
+		} else {
+			tp = types.NewFieldType(mysql.TypeNull)
 		}
 	case ast.Round:
-		t := x.Args[0].GetType().Tp
-		switch t {
-		case mysql.TypeBit, mysql.TypeTiny, mysql.TypeShort, mysql.TypeInt24, mysql.TypeLonglong:
-			tp = types.NewFieldType(mysql.TypeLonglong)
-		case mysql.TypeNewDecimal:
-			tp = types.NewFieldType(mysql.TypeNewDecimal)
-		default:
-			tp = types.NewFieldType(mysql.TypeDouble)
+		if len(x.Args) > 0 {
```

Comment:
I guess it can be put **before** the starting of switch statement, because every case statement do the check.